nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -65,44 +63,96 @@ impl DataStore {
                 &pagparams.map_name(|n| Name::ref_cast(n)),
             ),
         }
-        .filter(dsl::internal.eq(false))
+        // != excludes nulls so we explicitly include them
+        .filter(dsl::silo_id.ne(*INTERNAL_SILO_ID).or(dsl::silo_id.is_null()))
         .filter(dsl::time_deleted.is_null())
         .select(db::model::IpPool::as_select())
         .get_results_async(self.pool_authorized(opctx).await?)
         .await
         .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
     }
 
-    /// Looks up the default IP pool by name.
+    /// Looks up the default IP pool for a given scope, i.e., a given
+    /// combination of silo and project ID (or none). If there is no default at
+    /// a given scope, fall back up a level. There should always be a default at
+    /// fleet level, though this query can theoretically fail.
     pub async fn ip_pools_fetch_default_for(
         &self,
         opctx: &OpContext,
-        action: authz::Action,
-    ) -> LookupResult<(authz::IpPool, IpPool)> {
-        self.ip_pools_fetch_for(
-            opctx,
-            action,
-            &Name(ExternalName::from_str("default").unwrap()),
-        )
-        .await
+        silo_id: Option<Uuid>,
+        project_id: Option<Uuid>,
+    ) -> LookupResult<IpPool> {
+        use db::schema::ip_pool::dsl;
+
+        // TODO: Need auth check here. Only fleet viewers can list children on
+        // IP_POOL_LIST, so if we check that, nobody can make instances. This
+        // used to check CreateChild on an individual IP pool, but now we're not
+        // looking up by name so the check is more complicated
+        //
+        // opctx
+        //     .authorize(authz::Action::ListChildren, &authz::IP_POOL_LIST)
+        //     .await?;
```

Comment:
Interesting TODO. Not sure how to do this auth check given the way perms are set up for IpPool and IpPoolList:

https://github.com/oxidecomputer/omicron/blob/eee942f601bbebf00352eb5c5f060e1e80a437e6/nexus/db-queries/src/authz/omicron.polar#L368-L390

Reply:
Can we punt on this for now? Alternatively: if we don't do the most restrictive auth check right now are we concerned by the data exposed? Seems like a potentially reasonable punt.

Reply:
#3995